src/librustc_incremental/calculate_svh.rs

```diff
@@ -74,17 +36,24 @@ impl Svh {
         // to ensure it is not incorporating implementation artifacts into
         // the hash that are not otherwise visible.)
 
+        let crate_disambiguator = self.sess.crate_disambiguator.get();
+        let krate = self.map.krate();
+
         // FIXME: this should use SHA1, not SipHash. SipHash is not built to
         //        avoid collisions.
         let mut state = SipHasher::new();
+        debug!("state: {:?}", state);
 
         "crate_disambiguator".hash(&mut state);
-        crate_disambiguator.len().hash(&mut state);
-        crate_disambiguator.hash(&mut state);
+        crate_disambiguator.as_str().len().hash(&mut state);
```

Comment:
@michaelwoerister I guess we should discuss the role of the SVH. I see it as being used for two purposes:

1. to drive incr. comp.; for this role, endian-ness is no problem
2. to check for duplicate versions of a crate; here it's a bit less clear, endianness could be an issue in the sense that compiling the same source could produce distinct hashes on different host architectures.

I am somewhat...surprised to see that `write_usize` for hashes *does* seem to convert to a `[u8; 4]`. Crazy.

I guess we can write a "endian-proof" wrapper, though there is also code which converts `&[usize]` to `&[u8]`, which would be harder.